python/vyos/ifconfig/interface.py

```diff
@@ -1345,12 +1345,53 @@ def update(self, config):
 
         # create/update 802.1q VLAN interfaces
         for vif_id, vif_config in config.get('vif', {}).items():
+            
+            vif_ifname = f'{ifname}.{vif_id}'
+            vif_config['ifname'] = vif_ifname
+            
             tmp = deepcopy(VLANIf.get_config())
             tmp['source_interface'] = ifname
             tmp['vlan_id'] = vif_id
+            
+            if 'egress_qos' in vif_config:
+                egress_qos_str = ''
+                for qos_map in vif_config['egress_qos']:
+                    egress_qos_str += f'{qos_map} '
```

Comment:
Any reason to not use Python join() function?

Reply:
done